File: internals/order/controller/dto/order.go
package dto

import "time"

type Order struct {
	ID         string       `json:"id"`
	Code       string       `json:"code"`
	Lines      []*OrderLine `json:"lines"`
	TotalPrice float64      `json:"total_price"`
	Status     string       `json:"status"`
	UpdatedAt  time.Time    `json:"updated_at"`
}

type OrderLine struct {
	Product  Product `json:"product,omitempty"`
	Quantity uint    `json:"quantity"`
	Price    float64 `json:"price"`
}

type Product struct {
	ID          string  `json:"id"`
	Code        string  `json:"code"`
	Name        string  `json:"name"`
	Description string  `json:"description"`
	Price       float64 `json:"price"`
}

File: frontend/src/interface/order.ts
export interface Order {
  id: string
  code: string
  lines: OrderLine[]
  total_price: number
  status: string
  updated_at: string
}

export interface OrderLine {
  product: IProductOrder
  quantity: number
  price: number
}

export interface IProductOrder {
  id: string
  code: string
  name: string
  description: string
  price: number
}

export interface IListOrderRequest {
  code: string
  status: boolean
  page: number
  size: number
  order_by: string
  order_desc: boolean
}

export interface PlaceOrderRequest {
  user_id: string
  lines: PlaceOrderLineRequest[]
}

export interface PlaceOrderLineRequest {
  product_id: string
  quantity: number
}
